File: sbt/engine-job-api/src/main/java/sos/spooler/Job_impl.java
package sos.spooler;

public class Job_impl implements IJob_impl {

    public Log spooler_log;
    public Task spooler_task;
    public Job spooler_job;
    public Spooler spooler;

    protected Job_impl() {}

    @Override
    public final void initializeSpoolerVariables(Log log, Spooler spooler, Job job, Task task) {
        this.spooler_log = log;
        this.spooler = spooler;
        this.spooler_task = task;
        this.spooler_job = job;
    }

    @Override
    public boolean spooler_init() throws Exception {
        return true;
    }

    @Override
    public void spooler_exit() throws Exception {}

    @Override
    public boolean spooler_open() throws Exception {
        return true;
    }

    @Override
    public void spooler_close() throws Exception {}

    @Override
    public boolean spooler_process() throws Exception {
        return false;
    }

    @Override
    public void spooler_on_error() throws Exception {}

    @Override
    public void spooler_on_success() throws Exception {}
}

File: sbt/taskserver-dotnet/src/main/dotnet/adapter/Job_impl.cs
namespace sos.spooler
{
    public class Job_impl
    {
        public Log spooler_log { get; set; }
        public Task spooler_task { get; set; }
        public Job spooler_job { get; set; }
        public Spooler spooler { get; set; }

        #region JobScheduler API methods

        public virtual bool spooler_init()
        {
            return true;
        }

        public virtual bool spooler_open()
        {
            return true;
        }

        public virtual bool spooler_process()
        {
            return false;
        }

        public virtual void spooler_close()
        {
        }

        public virtual void spooler_on_success()
        {
        }

        public virtual void spooler_on_error()
        {
        }

        public virtual void spooler_exit()
        {
        }

        #endregion
    }
}
